From 7f4c724fa5a330b3108925c489168f69cc0707a0 Mon Sep 17 00:00:00 2001
From: Yang Shi <yang.shi@windriver.com>
Date: Mon, 16 Sep 2013 15:54:05 -0700
Subject: [PATCH] trace: Increase process name size for late boottime analysis

Add CONFIG_WR_BOOT kernel config to increase the buffer for the
process name so that analyse-boottime can display the process
name correctly.

Signed-off-by: Pradeep Tumati <pradeep.tumati@windriver.com>
[yshi: Add text for the config so that it can be showed in menuconfig]
Signed-off-by: Yang Shi <yang.shi@windriver.com>

diff --git a/include/linux/threads.h b/include/linux/threads.h
index 383ab9592bec..9e49628275e2 100644
--- a/include/linux/threads.h
+++ b/include/linux/threads.h
@@ -24,7 +24,11 @@
 /*
  * This controls the default maximum pid allocated to a process
  */
-#define PID_MAX_DEFAULT (CONFIG_BASE_SMALL ? 0x1000 : 0x8000)
+#define MAX_PID_NUM 0x8000
+#ifdef CONFIG_WR_BOOT
+#define MAX_PID_NUM 0x16000
+#endif
+#define PID_MAX_DEFAULT (CONFIG_BASE_SMALL ? 0x1000 : MAX_PID_NUM)
 
 /*
  * A maximum of 4 million PIDs should be enough for a while.
diff --git a/kernel/trace/Kconfig b/kernel/trace/Kconfig
index bbe95b9e0115..ab000c2ee6c6 100644
--- a/kernel/trace/Kconfig
+++ b/kernel/trace/Kconfig
@@ -330,6 +330,9 @@ config MISSED_TIMER_OFFSETS_HIST
 
 	      /sys/kernel/debug/tracing/latency_hist/enable/timerandwakeup
 
+config CONFIG_WR_BOOT
+	bool "Wind River boottime analysis"
+
 config ENABLE_DEFAULT_TRACERS
 	bool "Trace process context switches and events"
 	depends on !GENERIC_TRACER
diff --git a/kernel/trace/trace.c b/kernel/trace/trace.c
index 2e835e7f8b6a..05cc2e28442d 100644
--- a/kernel/trace/trace.c
+++ b/kernel/trace/trace.c
@@ -1229,7 +1229,11 @@ void tracing_reset_all_online_cpus(void)
 	}
 }
 
-#define SAVED_CMDLINES 128
+#define TOTAL_CMDLINE 128
+#ifdef CONFIG_WR_BOOT
+#define TOTAL_CMDLINE 256
+#endif
+#define SAVED_CMDLINES TOTAL_CMDLINE
 #define NO_CMDLINE_MAP UINT_MAX
 static unsigned map_pid_to_cmdline[PID_MAX_DEFAULT+1];
 static unsigned map_cmdline_to_pid[SAVED_CMDLINES];
-- 
1.9.0

